### ISSUE TITLE
Search appears broken on searches that yield results

### ISSUE THREAD

**Issue by explody** [gollum/gollum#939]
Discovered on gollum 3.1.1 w/ gollum-lib 4.0.1.  We had a running wiki that I just updated to current versions last night and now, searching for anything that returns a hit yields

```
NoMethodError - undefined method `nonzero?' for nil:NilClass:
	/usr/local/lib/ruby/gems/2.1.0/gems/gollum-3.1.1/lib/gollum/app.rb:430:in `block (2 levels) in <class:App>'
	/usr/local/lib/ruby/gems/2.1.0/gems/gollum-3.1.1/lib/gollum/app.rb:430:in `sort'
	/usr/local/lib/ruby/gems/2.1.0/gems/gollum-3.1.1/lib/gollum/app.rb:430:in `block in <class:App>'
```

To confirm, I did a fresh install of the gollum gem on my OSX machine, using the same git repo and with all default settings, and got the same results.  Oddly, searches that yield *no* hits succeed but obviously show no results.

I'll happily provide full stack traces, gem lists, etc. as requested.

Thanks!

**Comment by dometto** [comment]
I'm assuming that this is a duplicate of #923 -- if that's correct, this error should **not** occur if you have a page `foo` containing `foo` and you search for `foo`. If that does throw the error, please reopen this. I'll try to release the fix for #923 ASAP.

**Comment by explody** [comment]
I tried a page (literally) called 'foo' containing only 'foo' and searched for 'foo' but got the same result :\

**Comment by dometto** [comment]
:-( The similarity to #923 is striking though. Maybe my understanding of what caused #923 is incomplete. Are you able to test this with `gollum-lib HEAD` to see if that resolves the issue? If not, we'll have to do a bit more digging before doing a bugfix release.

**Comment by explody** [comment]
It definitely is very similar, I was surprised as well.  Yes, I'll update to HEAD and see what happens.

**Comment by explody** [comment]
Here I forked and was going to work on a fix but you beat me to it - fixed and working as expected with gollum-lib HEAD!

Thanks again

**Comment by dometto** [comment]
Glad to hear it!

**Comment by dometto** [comment]
Just released gollum-lib v4.0.2 in which this is fixed.
